File: src/algorithms/find-words-containing.js
/**
 * Given an array of strings words and a character x, returns an array of
 * indices representing the words that contain the character x.
 *
 * @format
 * @param {string[]} words - A array of strings made up of lowercase English
 *     letters
 * @param {string} x - A lowercase English letter
 * @returns {number[]} - Returns an array of indices where x appears in a word
 */

function findWordsContaining(words, x) {
  let indices = [];
  for (let i = 0; i < words.length; i++) {
    for (let j = 0; j < words[i].length; j++) {
      if (words[i].charAt(j) === x) {
        indices.push(i);
        break;
      }
    }
  }
  return indices;
}

File: src/algorithms/find-words-containing.py
def find_is_containing(words, x):
    """You are given a 0-indexed array of strings words and a character x
    Return an array of indices representing the words that contain the
    character x. Note that the returned array may be in any order.

    Args:
        words (List[str]): A list of strings.
        x (String): A lowercase English letter.

    Returns:
        List[int]: Returns a list of indices.
    """
    res = []
    index = 0
    for word in words:
        for char in word:
            if char == x:
                res.append(index)
                break
        index += 1
    return res
